File: frontend/src/models/Bill.ts
import type {BillItem} from "./BillItem";

export interface Bill {
    billNo: string;
    cashier: string;
    date: string;
    time: string;
    items: BillItem[];
    subTotal: number;
    serviceCharge: number;
    discount: number;
    total: number;
}

File: backend/src/main/java/com/ceylon_vistas/entity/Bill.java
package com.ceylon_vistas.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "Bills")
public class Bill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String billNo;

    private String date;
    private String time;
    private double subTotal;
    private double serviceCharge;
    private double discount;
    private double total;

    @OneToMany(mappedBy = "bill", cascade = CascadeType.ALL)
    private List<BillItem> items;
}